### PR TITLE
Add test for adding new page

### PR DESCRIPTION
**Pull request by rijju-pdl**
This PR implements test for adding new page.
```
Scenario: add new page

```

### CODE REVIEW

File: cucumber.conf.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ setDefaultTimeout(60000)
 BeforeAll(async function () {
   global.browser = await chromium.launch({
       headless: false,
-      slowMo: 1000,
+      slowMo: 100,
```
Comment (gurunglaxmi891) [inline cucumber.conf.js]:
```suggestion
      slowMo: 1000,
```

File: tests/features/addPage.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Feature: Login
```
Comment (gurunglaxmi891) [inline tests/features/addPage.feature]:
```suggestion
Feature: login
```

File: tests/features/addPage.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    As an admin
+    I want to add new page to the website
+    So that I can manage the website's products or contents
+
+        
+    Scenario: add new page
+        Given the user "admin" has logged in to the admin panel with email "admin@admin.com" and password "admin123456"
+        When the user "admin" adds a new page with the following details:
```
Comment (gurunglaxmi891) [inline tests/features/addPage.feature]:
```suggestion
        When user "admin" adds a new page with following details:
```

File: tests/pageObjects/AddPage.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class AddPage{
+  constructor() 
+  {
+    this.accountButtonSelector = "//button[contains(@id,'account-button')]";
+    // this.accountButtonSelector = "button#account-button";
```
Comment (gurunglaxmi891) [inline tests/pageObjects/AddPage.js]:
```suggestion
```

File: tests/pageObjects/AddPage.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    this.adminPanelSelector = "//div[contains(@id,'link-to-account')]//a[contains(@href,'/admin/orders')]";
```
Comment (gurunglaxmi891) [inline tests/pageObjects/AddPage.js]: Can you make this selector shorter?
```suggestion
    this.adminPanelSelector = "//div[contains(@id,'link-to-account')]//a[contains(@href,'/admin/orders')]";
```

File: tests/pageObjects/AddPage.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+   this.titleInputFieldSelector = "//input[@id='cms_page_title']";
+   this.pageTypeDropDownSelector = "//span[@id='select2-cms_page_type-container']";
+   this.pageSelector = "//li[text()='Feature Page']"
+   this.languageSelector = "//span[@id='select2-cms_page_locale-container']"
+   this.metaTitleSelector = "//input[@id='cms_page_meta_title']"
+   this.metaDescriptionSelector = "//textarea[@id='cms_page_meta_description']"
+   this.slugSelector = "//input[@id='cms_page_slug']"
+   this.createSelector = "//button[normalize-space()='Create']"
+   this.createdPageSelector = "//td[contains(text(),'%s')]"
+  }
```
Comment (gurunglaxmi891) [inline tests/pageObjects/AddPage.js]:
```suggestion
  }
  
```

File: tests/pageObjects/AddPage.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  async navigateToPageS() {
```
Comment (gurunglaxmi891) [inline tests/pageObjects/AddPage.js]:
```suggestion
  async navigateToPagesPage() {
```

File: tests/pageObjects/AddPage.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+class AddPage{
+  constructor() 
+  {
+    this.accountButtonSelector = "//button[contains(@id,'account-button')]";
+    // this.accountButtonSelector = "button#account-button";
+    this.adminPanelSelector = "//div[contains(@id,'link-to-account')]//a[contains(@href,'/admin/orders')]";
+   this.sidebarContentSelector ="a[href='#sidebar-content']";
+   this.sidebarPageSelector = "a[href='/admin/cms_pages']";
+   this.addNewPageButtonSelector = "//div[@class='page-actions pl-0 pr-0 d-none d-lg-flex']//a[@id='new_page']";
```
Comment (gurunglaxmi891) [inline tests/pageObjects/AddPage.js]: Can make this selector shorter?
```suggestion
   this.addNewPageButtonSelector = "//div[@class='page-actions pl-0 pr-0 d-none d-lg-flex']//a[@id='new_page']";
```

File: tests/pageObjects/LoginPage.js [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
- async navigateToAdminLoginPage() {
-    await page.goto("http://localhost:3000/admin/login");
- }
+    constructor() 
+    {
+        this.loginBtnSelector = "(//input[@name='commit'])[1]";
+        this.emailFieldSelector = "#spree_user_email";
+        this.passwordFieldSelector = "#spree_user_password";
+        this.errorMessageSelector = ".alert-danger";
+    }
```
Comment (gurunglaxmi891) [inline tests/pageObjects/LoginPage.js:8]:
```suggestion
    }
    
```

File: tests/stepDefinitions/addPageContext.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    await page.waitForURL("http://localhost:3000/", { timeout: 5000 });
+  });
+
+When('the user {string} adds a new page with the following details:', async function (user, pageDetails) {
+ await addPage.navigateToPageS();
+ const details = pageDetails.hashes();
+ await addPage.addNewPage(details[0].title, details[0].pageType, details[0].langauge, details[0].metaTitle, details[0].slug, details[0].metaDescription)
+})
+
+Then('the new page {string} should be listed in the pages list',async  function (createdpage) {
```
Comment (gurunglaxmi891) [inline tests/stepDefinitions/addPageContext.js]:
```suggestion
Then('the new page {string} should be listed in the pages list',async  function (createdPage) {
```

File: tests/features/addPage.feature [hunk 11]
```diff
@@ -0,0 +1,14 @@
+Feature: Login
+    As an admin
+    I want to add new page to the website
+    So that I can manage the website's products or contents
+
+        
+    Scenario: add new page
+        Given the user "admin" has logged in to the admin panel with email "admin@admin.com" and password "admin123456"
```
Comment (prashant-gurung899) [inline tests/features/addPage.feature]:
```suggestion
        Given user "admin" has logged in to the admin panel with email "admin@admin.com" and password "admin123456"
```

**Comment by koebel** [comment]
Does this PR relates to the issue about E2E tests (https://github.com/spree/spree/issues/12235)? Linking to issues helps other developers who are considering to make open source contributions see that someone is already working on this task and avoid wasting time and effort by having multiple people do the same task  🙈

And if there is any dependency of this Draft PR with another open PR towards the original project (maybe with https://github.com/spree/spree/pull/12267), it's also helpful to link them, in particular if you are considering to add your tests as an open source contribution towards the original project at some point in time in the future.

**Pull request merged**